hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTable.java

```diff
@@ -344,6 +345,26 @@ default CheckAndMutateBuilder ifEquals(byte[] value) {
     CompletableFuture<Boolean> thenMutate(RowMutations mutation);
   }
 
+  /**
+   * Batch checkAndMutates the specified data into the table.
+   *
+   * @param checkAndMutates The list of rows to apply.
+   * @return A list of {@link CompletableFuture}s that represent the result for each
+   *   CheckAndMutate.
+   */
+  List<CompletableFuture<Boolean>> checkAndMutate(List<CheckAndMutate> checkAndMutates);
+
+  /**
+   * A simple version of batch checkAndMutate. It will fail if there are any failures.
+   *
+   * @param checkAndMutates The list of rows to apply.
+   * @return A {@link CompletableFuture} that wrapper the result boolean list.
+   */
+  default CompletableFuture<List<Boolean>> checkAndMutateAll(
```

Comment:
Why would we offer this api?

Reply:
In AsyncTable, for methods returning List object, we provide 2 pattern methods returning List<CompletableFuture<T>> and CompletableFuture<List<T>>, and the latter's name is the former's name + "All". 

For example, we already provide AsyncTable#get and AsyncTable#getAll:
https://github.com/brfrn169/hbase/blob/f66cbe1a40cd9d4cd4abd27c7ac9d1a5168b885f/hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTable.java#L452-L471

I did the same for checkAndMutate and checkAndMutateAll.